protocol/messenger.go

```diff
@@ -2151,80 +2152,84 @@ func (m *Messenger) dispatchPairInstallationMessage(ctx context.Context, spec co
 	return id, nil
 }
 
-func (m *Messenger) dispatchMessage(ctx context.Context, spec common.RawMessage) (common.RawMessage, error) {
+func (m *Messenger) dispatchMessage(ctx context.Context, rawMessage common.RawMessage) (common.RawMessage, error) {
 	var err error
 	var id []byte
-	logger := m.logger.With(zap.String("site", "dispatchMessage"), zap.String("chatID", spec.LocalChatID))
-	chat, ok := m.allChats.Load(spec.LocalChatID)
+	logger := m.logger.With(zap.String("site", "dispatchMessage"), zap.String("chatID", rawMessage.LocalChatID))
+	chat, ok := m.allChats.Load(rawMessage.LocalChatID)
 	if !ok {
-		return spec, errors.New("no chat found")
+		return rawMessage, errors.New("no chat found")
 	}
 
 	switch chat.ChatType {
 	case ChatTypeOneToOne:
 		publicKey, err := chat.PublicKey()
 		if err != nil {
-			return spec, err
+			return rawMessage, err
 		}
 
 		//SendPrivate will alter message identity and possibly datasyncid, so we save an unchanged
 		//message for sending to paired devices later
-		specCopyForPairedDevices := spec
-		if !common.IsPubKeyEqual(publicKey, &m.identity.PublicKey) || spec.SkipEncryption {
-			id, err = m.sender.SendPrivate(ctx, publicKey, &spec)
+		specCopyForPairedDevices := rawMessage
+		if !common.IsPubKeyEqual(publicKey, &m.identity.PublicKey) || rawMessage.SkipEncryption {
+			id, err = m.sender.SendPrivate(ctx, publicKey, &rawMessage)
 
 			if err != nil {
-				return spec, err
+				return rawMessage, err
 			}
 		}
 
 		err = m.sendToPairedDevices(ctx, specCopyForPairedDevices)
 
 		if err != nil {
-			return spec, err
+			return rawMessage, err
 		}
 
 	case ChatTypePublic, ChatTypeProfile:
 		logger.Debug("sending public message", zap.String("chatName", chat.Name))
-		id, err = m.sender.SendPublic(ctx, chat.ID, spec)
+		id, err = m.sender.SendPublic(ctx, chat.ID, rawMessage)
 		if err != nil {
-			return spec, err
+			return rawMessage, err
 		}
 	case ChatTypeCommunityChat:
 		// TODO: add grant
 		canPost, err := m.communitiesManager.CanPost(&m.identity.PublicKey, chat.CommunityID, chat.CommunityChatID(), nil)
 		if err != nil {
-			return spec, err
+			return rawMessage, err
 		}
 
 		// We allow emoji reactions by anyone
-		if spec.MessageType != protobuf.ApplicationMetadataMessage_EMOJI_REACTION && !canPost {
+		if rawMessage.MessageType != protobuf.ApplicationMetadataMessage_EMOJI_REACTION && !canPost {
 			m.logger.Error("can't post on chat", zap.String("chat-id", chat.ID), zap.String("chat-name", chat.Name))
 
-			return spec, errors.New("can't post on chat")
+			return rawMessage, errors.New("can't post on chat")
 		}
 
 		logger.Debug("sending community chat message", zap.String("chatName", chat.Name))
-		id, err = m.sender.SendPublic(ctx, chat.ID, spec)
+		// TODO not public now
+		// id, err = m.sender.SendPublic(ctx, chat.ID, rawMessage)
+		decodedCommunityId, _ := types.DecodeHex(chat.CommunityID)
+
+		_, err = m.sender.SendCommunityMessage(ctx, decodedCommunityId, rawMessage)
```

Comment:
This looks like a breaking change or I am mistaking?

Reply:
Added a flag to Community Config - `Encrypted`, stored in `communities_communities` table. Hopefully this is a suitable location.